src/gui/guiChatConsole.cpp

```diff
@@ -773,6 +798,52 @@ bool GUIChatConsole::weblinkClick(s32 col, s32 row)
 	return false;
 }
 
+#ifdef USE_CURL
+// This is a function for CURL to write the page's html into a string
+size_t writeHtml(void* contents, size_t size, size_t nmemb, std::string* strptr) {
+    size_t totalSize = size * nmemb;
+    strptr->append((char*)contents, totalSize);
+    return totalSize;
```

Comment:
so you're downloading the entire page here? what happens if the server sends 100MB?

Reply:
What should be the limit? If the title is contained at the very beginning of the HTML, I agree there is no need to send the entire thing... Yet, when you link to a youtube video for example, it doesn't actually download the video, just HTML, which should normally be less than 1 Megabyte

Reply:
Now that I no longer using cURL myself though, (and let httpfetch take care of it), i lost the ability to stop the transfer.
Anyways, I tested a few webpages on the latest commit now, and I never got above one megabyte. https://bbc.com was 221 kilobytes, closest I've ever been was 950 kb with a YouTube video.